js/ui/discoveryFeedButton.js

```diff
@@ -18,13 +19,20 @@ function maybeCreateInactiveButton() {
     return null;
 }
 
+const DISCOVERY_FEED_PRIMARY_MONITOR_WIDTH_THRESHOLD = 1024;
+
+function _primaryMonitorWidthPassesThreshold() {
+    let primary = Gdk.Display.get_default().get_primary_monitor();
```

Comment:
You should not use `Gdk` to get the primary monitor size -- the shell already has this information, see `layout.js`.